components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AMDefaultKeyManagerImpl.java

```diff
@@ -339,16 +339,20 @@ public AccessTokenInfo getNewApplicationAccessToken(AccessTokenRequest tokenRequ
                 //Revoke the Old Access Token
                 httpRevokePost.setEntity(new UrlEncodedFormEntity(revokeParams, "UTF-8"));
                 int statusCode;
+                String responseBody;
                 try {
                     HttpResponse revokeResponse = executeHTTPrequest(revokeEndpointPort, revokeEndpointProtocol, 
                                                                      httpRevokePost);
                     statusCode = revokeResponse.getStatusLine().getStatusCode();
+                    responseBody = EntityUtils.toString(revokeResponse.getEntity());
                 } finally {
                     httpRevokePost.reset();
                 }
 
                 if (statusCode != 200) {
-                    throw new APIManagementException("Token revoke failed : HTTP error code : " + statusCode);
+                    String errorReason = "Token revoke failed : HTTP error code : " + statusCode +
+                                                                                             ". Reason " + responseBody;
+                    throw new APIManagementException(errorReason);
                 } else {
                     if (log.isDebugEnabled()) {
```

Comment:
Can't we remove log.isDebugEnabled() check since there is no string concatenation passed for log.debug method?

Reply:
+1 will fix later PR